esp-hal/src/peripherals.rs

```diff
@@ -204,6 +204,13 @@ for_each_peripheral! {
                 })
             }
 
+            /// Returns all the peripherals *once*.
+            #[inline]
+            #[cfg(not(feature = "rt"))]
+            pub(crate) fn take() -> Self {
+                crate::ESP_HAL_LOCK.lock(|| unsafe { Self::steal() })
+            }
+
```

Comment:
I don't think we should keep this. In fact, without "rt" the whole `Peripherals` struct should probably be removed.